srtcore/api.h

```diff
@@ -239,17 +244,70 @@ friend class CRendezvousQueue;
 
    CUDTException* getError();
 
+   CUDTGroup& addGroup(SRTSOCKET id)
+   {
+       srt::sync::CGuard cg (m_GlobControlLock);
+       // This only ensures that the element exists.
+       // If the element was newly added, it will be NULL.
+       CUDTGroup*& g = m_Groups[id];
+       if (!g)
+       {
+           // This is a reference to the cell, so it will
+           // rewrite it into the map.
+           g = new CUDTGroup;
+       }
+
+       // Now we are sure that g is not NULL,
+       // and persistence of this object is in the map.
+       // The reference to the object can be safely returned here.
+       return *g;
+   }
+
+   void deleteGroup(CUDTGroup* g)
+   {
+       using srt_logging::mglog;
+
+       srt::sync::CGuard cg (m_GlobControlLock);
+
+       CUDTGroup* pg = map_get(m_Groups, g->m_GroupID, NULL);
+       if (pg)
+           delete pg;
+       else
+       {
+           LOGC(mglog.Error, log << "IPE: the group id=" << g->m_GroupID << " not found in the map!");
+           // still delete it.
+           delete g;
+       }
+
+       m_Groups.erase(g->m_GroupID);
+   }
+
+   CUDTGroup* findPeerGroup(SRTSOCKET peergroup)
+   {
+       srt::sync::CGuard cg (m_GlobControlLock);
+
+       for (groups_t::iterator i = m_Groups.begin();
+               i != m_Groups.end(); ++i)
```

Comment:
alignment does not look nice here.

Reply:
Why not. It's a double indent, like always when parenthesis-enclosed expression is split int multiple lines. Do we have something else in the formatting configuration?